### PR TITLE
🤖 Updated get_vote_sum, and added tests

### PR DESCRIPTION
**Pull request by phil-r**

### CODE REVIEW

**Comment by phil-r** [comment]
ow, I forgot

File: tests/github_api/voting_test.py [hunk 1]
```diff
@@ -29,3 +29,27 @@ def test_get_approval_threshold(self, mock_get_num_watchers):
...
+        self.assertEqual(voting.get_vote_sum('fake_api', mocked_vote_result), expected_vote_sum)
+
+        # all positive example
+        mocked_vote_result = {"a": 1, "b": 1, "c": 1, "d": 1, "x": 1, "y": 1, "z": 1}
+        expected_vote_sum = (7, 0)
+        self.assertEqual(voting.get_vote_sum('fake_api', mocked_vote_result), expected_vote_sum)
+
+        # almost all negative example, level of controversy is low
+        mocked_vote_result = {"a": 1, "b": -1, "c": -1, "d": -1, "x": -1, "y": -1, "z": -1}
+        expected_vote_sum = (-5, 1)
```
Comment (phil-r) [inline tests/github_api/voting_test.py:49]: before my changes `variance` for this test was 6, and that's ~~bullshit~~ incorrect

**Comment by yet-another-account** [comment]
Don't forget to add "[almost arrested!] [gone sexual] [you wont believe what happens!!1] [CRAZY BUT TRUE!!!] [MUST READ] [INSANE CHALLENGE!]" to the title.

File: github_api/voting.py [hunk 2]
```diff
@@ -125,16 +125,21 @@ def get_vote_weight(api, username):
...
-
-    return total, (variance - total)
+        weighted_vote = weight * vote
+        total += weighted_vote
+        if weighted_vote > 0:
+            positive += weighted_vote
+        elif weighted_vote < 0:
+            negative -= weighted_vote
+
+    variance = min(positive, negative)
```
Comment (rudehn) [inline github_api/voting.py:141]: Do you want an abs(negative) here?

Comment (phil-r) [reply github_api/voting.py:141]: I made a clever trick on line 139 xD

Comment (rudehn) [reply github_api/voting.py:141]: ahh you did at that!

**Comment by phil-r** [comment]
@eukaryote31 I'm trying my best

**Comment by mark-i-m** [comment]
The title is getting too long... how about

`🤖 [THIS PR WILL CHANGE YOUR...`

File: github_api/voting.py (hunk 2 above)
Comment (yet-another-account) [reply github_api/voting.py:141]: It's sort of unreadable, why not `negative *= -1` or something?

**Comment by yet-another-account** [comment]
No, put the clickbait at the beginning, and move all the relevant stuff to the very end

File: github_api/voting.py (hunk 2 above)
Comment (phil-r) [reply github_api/voting.py:141]: I can do `negative += weighted_vote` and then `variance = min(positive, abs(negative))` will it be more readable?

**Pull request merged**